pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -160,18 +184,29 @@ protected synchronized void deleteSegmentFromPropertyStoreAndLocal(String tableN
     if (!segmentsToRetryLater.isEmpty()) {
       long effectiveDeletionDelay = Math.min(deletionDelay * 2, MAX_DELETION_DELAY_SECONDS);
       LOGGER.info("Postponing deletion of {} segments from table {}", segmentsToRetryLater.size(), tableName);
-      deleteSegmentsWithDelay(tableName, segmentsToRetryLater, effectiveDeletionDelay);
+      deleteSegmentsWithDelay(tableName, segmentsToRetryLater, deletedSegmentsRetentionMs, effectiveDeletionDelay);
       return;
     }
   }
 
   public void removeSegmentsFromStore(String tableNameWithType, List<String> segments) {
+    removeSegmentsFromStore(tableNameWithType, segments, _defaultDeletedSegmentsRetentionMs, true);
+  }
+
+  public void removeSegmentsFromStore(String tableNameWithType, List<String> segments,
+      long deletedSegmentsRetentionMs) {
+    removeSegmentsFromStore(tableNameWithType, segments, deletedSegmentsRetentionMs, false);
+  }
+
+  public void removeSegmentsFromStore(String tableNameWithType, List<String> segments,
+      long deletedSegmentsRetentionMs, boolean usedDefaultClusterRetention) {
```

Comment:
We don't need `usedDefaultClusterRetention` if we change `deletedSegmentsRetentionMs` to `@Nullable Long`